### PR TITLE
Add Yandex OAuth2 provider (#8335)

### LINKED ISSUE #8335: [FR] add Yandex as oauth2 support
- Gitea version (or commit ref): 
- Git version:
- Operating system: 
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

[goth](https://github.com/markbates/goth) already supports Yandex. It would be great to see Yandex as oauth2 provider as many organizations (including mine) already usages Yandex.

---

I wish I was familiar with Golang, as looking at #4476 it doesn't seem to complicated, though that could be my lack of experience. Would be a really great see this implemented.

### PR DESCRIPTION
**Pull request by anatolym**
Fix #8335

Screen Shot 2020-03-01 at 17 55 32

### CODE REVIEW

File: options/locale/locale_ru-RU.ini [hunk 1]
```diff
@@ -1774,7 +1774,8 @@ auths.tip.google_plus=Получите учетные данные клиент
 auths.tip.openid_connect=Используйте OpenID Connect Discovery URL (<server>/.well-known/openid-configuration) для автоматической настройки входа OAuth
 auths.tip.twitter=Перейдите на https://dev.twitter.com/apps, создайте приложение и убедитесь, что включена опция «Разрешить это приложение для входа в систему с помощью Twitter»
 auths.tip.discord=Добавьте новое приложение на https://discordapp.com/developers/applications/me
-auths.tip.gitea=Зарегистрировать новое приложение OAuth2. Руководство можно найти на https://docs.gitea.io/ru-us/oauth2-provider/
+auths.tip.gitea=Зарегистрируйте новое приложение OAuth2. Руководство можно найти на https://docs.gitea.io/ru-us/oauth2-provider/
```
Comment (guillep2k) [inline options/locale/locale_ru-RU.ini]: Please revert changes in locales different from `en-US`. These locales are updated via [Crowdin](https://crowdin.com/project/gitea).

**Review by guillep2k — APPROVED** [review]

File: modules/auth/oauth2/oauth2.go [hunk 2]
```diff
@@ -24,7 +24,8 @@ import (
 	"github.com/markbates/goth/providers/google"
 	"github.com/markbates/goth/providers/openidConnect"
 	"github.com/markbates/goth/providers/twitter"
-	"github.com/satori/go.uuid"
+	"github.com/markbates/goth/providers/yandex"
+	uuid "github.com/satori/go.uuid"
 	"xorm.io/xorm"
```
Comment (6543) [inline modules/auth/oauth2/oauth2.go:30]:
```suggestion

	"xorm.io/xorm"
```

Comment (6543) [reply modules/auth/oauth2/oauth2.go:30]: Was not formated before but would be nice if you include this :)

**Review by 6543 — APPROVED** [review]

**Comment by anatolym** [comment]
Looks like some stuff keeps failing. Is it something I should address?

**Comment by techknowlogick** [comment]
Seems when I resolved conflicts I didn't do it in a way that was go fmt capable, could you run `make fmt`?

File: templates/admin/auth/new.tmpl [hunk 3]
```diff
@@ -115,8 +115,10 @@
 				<span>{{.i18n.Tr "admin.auths.tip.discord"}}</span>
 				<li>Gitea</li>
 				<span>{{.i18n.Tr "admin.auths.tip.gitea"}}</span>
-				<li>Nextcloud</li>
+        <li>Nextcloud</li>
```
Comment (lafriks) [inline templates/admin/auth/new.tmpl]:
```suggestion
				<li>Nextcloud</li>
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by lafriks — APPROVED** [review]

File: modules/auth/oauth2/oauth2.go [hunk 4]
```diff
@@ -25,7 +25,9 @@ import (
 	"github.com/markbates/goth/providers/nextcloud"
 	"github.com/markbates/goth/providers/openidConnect"
 	"github.com/markbates/goth/providers/twitter"
-	"github.com/satori/go.uuid"
+	"github.com/markbates/goth/providers/yandex"
+	uuid "github.com/satori/go.uuid"
+
```
Comment (lafriks) [inline modules/auth/oauth2/oauth2.go]: Not needed newline and no need for import alias

**Review by lafriks — CHANGES_REQUESTED** [review]

File: modules/auth/oauth2/oauth2.go (hunk 2 above)
Comment (6543) [reply modules/auth/oauth2/oauth2.go:30]: @lafriks dont add a space there?

Comment (guillep2k) [reply modules/auth/oauth2/oauth2.go:30]: @6543 "xorm.io/xorm" is an external library, same as "github.com/satori/go.uuid", so they go grouped together. There are only three groups: golang's, "code/gitea.io/gitea" and the rest.

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
Looks good now

**Pull request merged**